### ISSUE TITLE
torch.distributed.elastic.multiprocessing.api: failed exitcode 1 local rand 1

### ISSUE THREAD

**Issue by Yingyue-L** [luogen1996/LaVIN#2]
Thanks for your work! I try to reproduce the finetuning_sqa_7b scripts on 4x3090Ti that set batch_size=1, but I meet an error when I run 770 iters as show below. Can you give the version of your packages?
![image](https://github.com/luogen1996/LaVIN/assets/56422335/932b1860-671b-4f82-9fc7-938200736173)

**Comment by Yingyue-L** [comment]
Thanks for your answer! I fix the error by set the pytorch version ==1.10.0+cu113.

**Comment by Yingyue-L** [comment]
I meet a load weight mismatching error when I try to evaluate on multi-gpus, can you help me ?

**Comment by luogen1996** [comment]
> I meet a load weight mismatching error when I try to evaluate on multi-gpus, can you help me ?

Multi-gpu validation is not supported now, and single-gpu will be fine.

**Comment by Yingyue-L** [comment]
![image](https://github.com/luogen1996/LaVIN/assets/56422335/b7c86195-c497-400a-8221-1b5adf9d512f)
![image](https://github.com/luogen1996/LaVIN/assets/56422335/e934b392-7803-4d58-ad7d-9601e6c10ac7)
Is there a writing error on your paper? What is the real performance for LaVIN-7B?

**Comment by luogen1996** [comment]
The performance of LaVIN-7B is 89.4, we will revise the typos in our paper.

**Comment by Yingyue-L** [comment]
> The performance of LaVIN-7B is 89.4, we will revise the typos in our paper.

Thanks. I find that when set gradient checkpointing for MMAdapter and CLIP-Adapter, it can run with batchsize=4 on 3090Ti. But the Acc_average on ScienceQA test is only 88.75, is there any problem?

**Comment by Yingyue-L** [comment]
I meet the Loss nan when try to use vicuna-13b, can you help?

**Comment by luogen1996** [comment]
> I meet the Loss nan when try to use vicuna-13b, can you help?

You'd better to provide the training setups like batch size etc.

**Comment by Yingyue-L** [comment]
> > I meet the Loss nan when try to use vicuna-13b, can you help?
> 
> You'd better to provide the training setups like batch size etc.

```bash
CUDA_VISIBLE_DEVICES=0,1,2,3 torchrun --nproc_per_node 4 --master_port 16598 train.py \
    --llm_model 7B \
    --llama_model_path ../data/weights/ \
    --data_path ../data/alpaca_data.json \
    --max_seq_len 512 \
    --batch_size 4 \
    --accum_iter 2 \
    --epochs 20 \
    --warmup_epochs 2 \
    --blr 9e-3 \
    --weight_decay 0.02 \
    --output_dir ./LaVIN-Vicuna-7B/ \
    --log_dir ./LaVIN-Vicuna-7B/ \
    --adapter_type attn \
    --adapter_dim 8 \
    --adapter_scale 1 \
    --n_prompt 6 \
    --prompt_format QCM-ALE \
    --temperature 10. \
    --visual_adapter_type router \
    --use_vicuna \
    --gradient_checkpointing
```
I use the provided scripts, and just add the gradient_clipping

**Comment by zhijiew** [comment]
> I meet the Loss nan when try to use vicuna-13b, can you help?

Same problem here. Using LLaMA-13B for fine-tuning is ok, but the “loss is nan” error occurs when I load the Vicuna-13B weight.

**Comment by luogen1996** [comment]
If you meet the same problem, please replace line 48-50 in engine.py with the following codes: 
`       
 if torch.isnan(loss):
            print("NaN loss encountered. Skipping this batch.")
            continue
`
Our codes will be updated soon, and this problem will be well addressed. Thanks for your interest.
